src/Sulu/Bundle/ContactBundle/Contact/ContactManager.php

```diff
@@ -195,7 +196,20 @@ public function setMainAccount(ContactEntity $contact, $data)
         } else {
             // if a main account exists - remove it
             if ($accountContact = $this->getMainAccountContact($contact)) {
+                
+                $qb = $this->em->createQueryBuilder();
```

Comment:
you should not create queries in the manager, this is what the repository is good for.